src/compiler/response_file.rs

```diff
@@ -0,0 +1,165 @@
+/// An iterator over the arguments in a response file.
```

Comment:
i would prefer the move to be done into a single commit 
and then, modifications into a second commit

Reply:
Sure, I'll rewrite the commit into two commits in that order.

Reply:
@Alexei-Barnes ping ? :)

Reply:
Sorry for the slow action here - I've been pulled into a work project that's eating my free time. I should get back to this soon.

Reply:
are you still going to work on this? thanks

Reply:
Right, sorry for the mega wait - I'm working on this again now.

Reply:
OK, this is split into two separate commits now. The first commit just moves the code, with minimal changes necessary to do so. The second change modifies it, adds tests, and uses it in the GCC module.

Reply:
I've now also merged main into this branch, to keep it up to date. No merge conflicts.